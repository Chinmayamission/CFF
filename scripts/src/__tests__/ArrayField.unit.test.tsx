import React from "react";
import { shallow, mount, render } from "enzyme";
import CustomForm from "../form/CustomForm";

it("renders regular checkboxes correctly", () => {
  let schema = {
    type: "object",
    title: "Form",
    properties: {
      interests: {
        title: "Title interests Tell us about your interests",
        description: "Description interests blah blah blah",
        type: "array",
        uniqueItems: true,
        items: {
          type: "string",
          enum: [
            "Art",
            "Singing",
            "Music",
            "Dance",
            "Sports",
            "Cooking",
            "Writing",
            "Coding"
          ]
        }
      }
    }
  };
  let uiSchema = {
    interests: {
      "ui:widget": "checkboxes"
    }
  };
  const wrapper = render(<CustomForm schema={schema} uiSchema={uiSchema} />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.text()).toContain(
    "Title interests Tell us about your interests"
  );
  expect(wrapper.text()).toContain("Description interests blah blah blah");
});

it("renders regular object array", () => {
  let schema = require("./schemas/parentSchema.json");
  let uiSchema = {};
  const wrapper = render(<CustomForm schema={schema} uiSchema={uiSchema} />);
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.text()).toContain("Add");
  expect(wrapper.text()).not.toContain("Remove");
});

it("renders object array with expand to maximum (two parents)", () => {
  let schema = require("./schemas/parentSchema.json");
  let uiSchema = {};
  let defaultFormData = { parents: [{}, {}] };
  const wrapper = render(
    <CustomForm
      schema={schema}
      uiSchema={uiSchema}
      formData={defaultFormData}
    />
  );
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.text()).not.toContain("Add");
  expect(wrapper.text()).toContain("Remove");
});

it("shows add button when minItems is zero", () => {
  let schema = {
    type: "object",
    properties: {
      children: {
        type: "array",
        items: {
          type: "string",
          title: "My Item Here"
        }
      }
    }
  };
  let uiSchema = {};
  let defaultFormData = {};
  const wrapper = render(
    <CustomForm
      schema={schema}
      uiSchema={uiSchema}
      formData={defaultFormData}
    />
  );
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.text()).not.toContain("My Item Here");
  expect(wrapper.text()).toContain("Add");
});

it("shows one child which can be removed", () => {
  let schema = {
    type: "object",
    properties: {
      children: {
        type: "array",
        items: {
          type: "string",
          title: "My Item Here"
        }
      }
    }
  };
  let uiSchema = {};
  let defaultFormData = { children: [{}] };
  const wrapper = render(
    <CustomForm
      schema={schema}
      uiSchema={uiSchema}
      formData={defaultFormData}
    />
  );
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.text()).toContain("My Item Here");
  expect(wrapper.text()).toContain("Add");
  expect(wrapper.text()).toContain("Remove");
});

it("custom title for one and two", () => {
  let schema = {
    type: "object",
    properties: {
      parents: {
        type: "array",
        minItems: 2,
        items: {
          type: "string"
        }
      }
    }
  };
  let uiSchema = {
    parents: {
      "ui:cff:arrayItemTitles": ["Parent Info", "Spouse Info"]
    }
  };
  let defaultFormData = {};
  const wrapper = render(
    <CustomForm
      schema={schema}
      uiSchema={uiSchema}
      formData={defaultFormData}
    />
  );
  expect(wrapper).toMatchSnapshot();
  expect(wrapper.text()).toContain("Parent Info");
  expect(wrapper.text()).toContain("Spouse Info");
});
