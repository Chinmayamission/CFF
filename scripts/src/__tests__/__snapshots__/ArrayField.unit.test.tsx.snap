// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders object array with expand to maximum (parents) 1`] = `
<div
  class="ccmt-cff-Page-FormPage"
>
  <form
    class="rjsf"
    novalidate=""
  >
    <div
      class="form-group field field-object"
    >
      <div
        class="container-fluid"
      >
        <h2
          class="ccmt-cff-form-title"
        >
          Form
        </h2>
        <div />
        <div
          class="row"
        >
          <div
            class="form-group field field-array  col-12"
          >
            <fieldset
              class="field field-array field-array-of-object"
            >
              <h2
                class="ccmt-cff-form-title"
              >
                parents
              </h2>
              <div
                class="array-item-list"
              >
                <div
                  class="row mb-4"
                >
                  <div
                    class="col-9"
                  >
                    <div
                      class="form-group field field-object  col-12"
                    >
                      <div
                        class="container-fluid"
                      >
                        <span />
                        <div />
                        <div
                          class="row"
                        >
                          <div
                            class="form-group field field-boolean  col-12"
                          >
                            <div
                              class="checkbox form-check "
                            >
                              <input
                                class="form-check-input"
                                id="root_parents_0_previous"
                                style="bottom:0"
                                type="checkbox"
                              />
                              <label
                                class="form-check-label"
                                for="root_parents_0_previous"
                              />
                            </div>
                            <div />
                            <div />
                          </div>
                          <div
                            class="form-group field field-string  col-12"
                          >
                            <label
                              class="control-label"
                              for="root_parents_0_volunteer"
                            >
                              Would you like to volunteer at CMA Marietta? If so, how?
                            </label>
                            <select
                              class="form-control"
                              id="root_parents_0_volunteer"
                            >
                              <option
                                selected=""
                                value=""
                              />
                              <option
                                value="Teaching"
                              >
                                Teaching
                              </option>
                              <option
                                value="Backup teacher"
                              >
                                Backup teacher
                              </option>
                              <option
                                value="Art activity"
                              >
                                Art activity
                              </option>
                              <option
                                value="Music"
                              >
                                Music
                              </option>
                              <option
                                value="Dance"
                              >
                                Dance
                              </option>
                              <option
                                value="Technology"
                              >
                                Technology
                              </option>
                              <option
                                value="Cooking"
                              >
                                Cooking
                              </option>
                              <option
                                value="Event organization"
                              >
                                Event organization
                              </option>
                            </select>
                            <div />
                            <div />
                          </div>
                          <div
                            class="form-group field field-object  col-12"
                          >
                            <div
                              class="container-fluid"
                            >
                              <span />
                              <div />
                              <div
                                class="row"
                              >
                                <div
                                  class="form-group field field-string  col-12"
                                >
                                  <label
                                    class="control-label"
                                    for="root_parents_0_name_last"
                                  >
                                    Last Name
                                    <span
                                      class="required"
                                    >
                                      *
                                    </span>
                                  </label>
                                  <input
                                    class="form-control"
                                    id="root_parents_0_name_last"
                                    label="Last Name"
                                    placeholder=""
                                    required=""
                                    type="text"
                                    value=""
                                  />
                                  <div />
                                  <div />
                                </div>
                                <div
                                  class="form-group field field-string  col-12"
                                >
                                  <label
                                    class="control-label"
                                    for="root_parents_0_name_first"
                                  >
                                    First Name
                                    <span
                                      class="required"
                                    >
                                      *
                                    </span>
                                  </label>
                                  <input
                                    class="form-control"
                                    id="root_parents_0_name_first"
                                    label="First Name"
                                    placeholder=""
                                    required=""
                                    type="text"
                                    value=""
                                  />
                                  <div />
                                  <div />
                                </div>
                              </div>
                            </div>
                            <div />
                            <div />
                          </div>
                          <div
                            class="form-group field field-string  col-12"
                          >
                            <label
                              class="control-label"
                              for="root_parents_0_phone"
                            >
                              Phone Number
                            </label>
                            <input
                              class="form-control"
                              id="root_parents_0_phone"
                              label="Phone Number"
                              placeholder=""
                              type="text"
                              value=""
                            />
                            <div />
                            <div />
                          </div>
                          <div
                            class="form-group field field-string  col-12"
                          >
                            <label
                              class="control-label"
                              for="root_parents_0_email"
                            >
                              email
                              <span
                                class="required"
                              >
                                *
                              </span>
                            </label>
                            <input
                              class="form-control"
                              id="root_parents_0_email"
                              label="email"
                              placeholder=""
                              required=""
                              type="email"
                              value=""
                            />
                            <div />
                            <div />
                          </div>
                        </div>
                      </div>
                      <div />
                      <div />
                    </div>
                  </div>
                  <div
                    class="col-3 ccmt-cff-array-button-container"
                  >
                    <button
                      class="btn btn-danger col-12 ccmt-cff-btn-array-remove"
                      style="visibility:hidden"
                      type="button"
                    >
                      Remove
                    </button>
                    <button
                      class="btn btn-info col-12 ccmt-cff-btn-array-add"
                      type="button"
                    >
                      Add
                    </button>
                  </div>
                </div>
              </div>
            </fieldset>
            <div />
            <div />
          </div>
        </div>
      </div>
      <div />
      <div />
    </div>
    <p>
      <button
        class="btn btn-info"
        type="submit"
      >
        Submit
      </button>
    </p>
  </form>
</div>
`;

exports[`renders regular checkboxes correctly 1`] = `
<div
  class="ccmt-cff-Page-FormPage"
>
  <form
    class="rjsf"
    novalidate=""
  >
    <div
      class="form-group field field-object"
    >
      <div
        class="container-fluid"
      >
        <h2
          class="ccmt-cff-form-title"
        >
          Form
        </h2>
        <div />
        <div
          class="row"
        >
          <div
            class="form-group field field-array  col-12"
          >
            <label
              class="control-label"
              for="root_interests"
            >
              Title interests Tell us about your interests
            </label>
            <div
              class="my-2"
              id="root_interests__description"
            >
              <div>
                Description interests blah blah blah
              </div>
            </div>
            <div
              class="checkboxes"
              id="root_interests"
            >
              <div
                class="checkbox "
              >
                <label>
                  <span>
                    <input
                      id="root_interests_0"
                      type="checkbox"
                    />
                    <span>
                      Art
                    </span>
                  </span>
                </label>
              </div>
              <div
                class="checkbox "
              >
                <label>
                  <span>
                    <input
                      id="root_interests_1"
                      type="checkbox"
                    />
                    <span>
                      Singing
                    </span>
                  </span>
                </label>
              </div>
              <div
                class="checkbox "
              >
                <label>
                  <span>
                    <input
                      id="root_interests_2"
                      type="checkbox"
                    />
                    <span>
                      Music
                    </span>
                  </span>
                </label>
              </div>
              <div
                class="checkbox "
              >
                <label>
                  <span>
                    <input
                      id="root_interests_3"
                      type="checkbox"
                    />
                    <span>
                      Dance
                    </span>
                  </span>
                </label>
              </div>
              <div
                class="checkbox "
              >
                <label>
                  <span>
                    <input
                      id="root_interests_4"
                      type="checkbox"
                    />
                    <span>
                      Sports
                    </span>
                  </span>
                </label>
              </div>
              <div
                class="checkbox "
              >
                <label>
                  <span>
                    <input
                      id="root_interests_5"
                      type="checkbox"
                    />
                    <span>
                      Cooking
                    </span>
                  </span>
                </label>
              </div>
              <div
                class="checkbox "
              >
                <label>
                  <span>
                    <input
                      id="root_interests_6"
                      type="checkbox"
                    />
                    <span>
                      Writing
                    </span>
                  </span>
                </label>
              </div>
              <div
                class="checkbox "
              >
                <label>
                  <span>
                    <input
                      id="root_interests_7"
                      type="checkbox"
                    />
                    <span>
                      Coding
                    </span>
                  </span>
                </label>
              </div>
            </div>
            <div />
            <div />
          </div>
        </div>
      </div>
      <div />
      <div />
    </div>
    <p>
      <button
        class="btn btn-info"
        type="submit"
      >
        Submit
      </button>
    </p>
  </form>
</div>
`;

exports[`renders regular object array 1`] = `
<div
  class="ccmt-cff-Page-FormPage"
>
  <form
    class="rjsf"
    novalidate=""
  >
    <div
      class="form-group field field-object"
    >
      <div
        class="container-fluid"
      >
        <h2
          class="ccmt-cff-form-title"
        >
          Form
        </h2>
        <div />
        <div
          class="row"
        >
          <div
            class="form-group field field-array  col-12"
          >
            <fieldset
              class="field field-array field-array-of-object"
            >
              <h2
                class="ccmt-cff-form-title"
              >
                parents
              </h2>
              <div
                class="array-item-list"
              >
                <div
                  class="row mb-4"
                >
                  <div
                    class="col-9"
                  >
                    <div
                      class="form-group field field-object  col-12"
                    >
                      <div
                        class="container-fluid"
                      >
                        <span />
                        <div />
                        <div
                          class="row"
                        >
                          <div
                            class="form-group field field-boolean  col-12"
                          >
                            <div
                              class="checkbox form-check "
                            >
                              <input
                                class="form-check-input"
                                id="root_parents_0_previous"
                                style="bottom:0"
                                type="checkbox"
                              />
                              <label
                                class="form-check-label"
                                for="root_parents_0_previous"
                              />
                            </div>
                            <div />
                            <div />
                          </div>
                          <div
                            class="form-group field field-string  col-12"
                          >
                            <label
                              class="control-label"
                              for="root_parents_0_volunteer"
                            >
                              Would you like to volunteer at CMA Marietta? If so, how?
                            </label>
                            <select
                              class="form-control"
                              id="root_parents_0_volunteer"
                            >
                              <option
                                selected=""
                                value=""
                              />
                              <option
                                value="Teaching"
                              >
                                Teaching
                              </option>
                              <option
                                value="Backup teacher"
                              >
                                Backup teacher
                              </option>
                              <option
                                value="Art activity"
                              >
                                Art activity
                              </option>
                              <option
                                value="Music"
                              >
                                Music
                              </option>
                              <option
                                value="Dance"
                              >
                                Dance
                              </option>
                              <option
                                value="Technology"
                              >
                                Technology
                              </option>
                              <option
                                value="Cooking"
                              >
                                Cooking
                              </option>
                              <option
                                value="Event organization"
                              >
                                Event organization
                              </option>
                            </select>
                            <div />
                            <div />
                          </div>
                          <div
                            class="form-group field field-object  col-12"
                          >
                            <div
                              class="container-fluid"
                            >
                              <span />
                              <div />
                              <div
                                class="row"
                              >
                                <div
                                  class="form-group field field-string  col-12"
                                >
                                  <label
                                    class="control-label"
                                    for="root_parents_0_name_last"
                                  >
                                    Last Name
                                    <span
                                      class="required"
                                    >
                                      *
                                    </span>
                                  </label>
                                  <input
                                    class="form-control"
                                    id="root_parents_0_name_last"
                                    label="Last Name"
                                    placeholder=""
                                    required=""
                                    type="text"
                                    value=""
                                  />
                                  <div />
                                  <div />
                                </div>
                                <div
                                  class="form-group field field-string  col-12"
                                >
                                  <label
                                    class="control-label"
                                    for="root_parents_0_name_first"
                                  >
                                    First Name
                                    <span
                                      class="required"
                                    >
                                      *
                                    </span>
                                  </label>
                                  <input
                                    class="form-control"
                                    id="root_parents_0_name_first"
                                    label="First Name"
                                    placeholder=""
                                    required=""
                                    type="text"
                                    value=""
                                  />
                                  <div />
                                  <div />
                                </div>
                              </div>
                            </div>
                            <div />
                            <div />
                          </div>
                          <div
                            class="form-group field field-string  col-12"
                          >
                            <label
                              class="control-label"
                              for="root_parents_0_phone"
                            >
                              Phone Number
                            </label>
                            <input
                              class="form-control"
                              id="root_parents_0_phone"
                              label="Phone Number"
                              placeholder=""
                              type="text"
                              value=""
                            />
                            <div />
                            <div />
                          </div>
                          <div
                            class="form-group field field-string  col-12"
                          >
                            <label
                              class="control-label"
                              for="root_parents_0_email"
                            >
                              email
                              <span
                                class="required"
                              >
                                *
                              </span>
                            </label>
                            <input
                              class="form-control"
                              id="root_parents_0_email"
                              label="email"
                              placeholder=""
                              required=""
                              type="email"
                              value=""
                            />
                            <div />
                            <div />
                          </div>
                        </div>
                      </div>
                      <div />
                      <div />
                    </div>
                  </div>
                  <div
                    class="col-3 ccmt-cff-array-button-container"
                  >
                    <button
                      class="btn btn-danger col-12 ccmt-cff-btn-array-remove"
                      style="visibility:hidden"
                      type="button"
                    >
                      Remove
                    </button>
                    <button
                      class="btn btn-info col-12 ccmt-cff-btn-array-add"
                      type="button"
                    >
                      Add
                    </button>
                  </div>
                </div>
              </div>
            </fieldset>
            <div />
            <div />
          </div>
        </div>
      </div>
      <div />
      <div />
    </div>
    <p>
      <button
        class="btn btn-info"
        type="submit"
      >
        Submit
      </button>
    </p>
  </form>
</div>
`;
